docs/admin/etcd.md

```diff
@@ -1,49 +1,163 @@
 ---
 assignees:
-- lavalamp
-title: Configuring Kubernetes Use of etcd
+- mml
+- wojtek-t
+title: Upgrading and Rolling Back etcd
 ---
 
+## About etcd
+
 [etcd](https://coreos.com/etcd/docs/latest/) is a highly-available key value
 store which Kubernetes uses for persistent storage of all of its REST API
-objects.
+objects.  For the mechanics behind how kubernetes builds, distributes and
+deploys Kubernetes, see _some doc_.
+
+## Background
+
+As of 1.5.1 kubernetes release, we are still using etcd from 2.2.1 release with
+v2 API.  Also, we have no pre-existing process for updating etcd, as we have
+never updated etcd by either minor or major version.
+
+Note that we need to migrate both: etcd versions that we are using (from 2.2.1
+to at least 3.0.x) as well as the version of API Kubernetes talks to etcd (3.0.x
+etcd binaries support both v2 and v3 API).
+
+This document describes how to do this migration.
+
+### Etcd upgrade limitations
+
+The way etcd was designed introduces some significant limitations on how the
+upgrade can be done. These are the main limitations. 
+
+#### One minor release at a time
+
+It is possible to upgrade only by one minor release at a time (though, within
+patch releases you can switch versions at any time back and forth). That
+basically means, that we cannot upgrade directly e.g. from 2.1.x to 2.3.x.
+
+Fortunately, this limitation it is easy to workaround - it is enough to start a
```

Comment:
The comma belongs, but I'll defer to a "doc review" at some point.

See "sentence adverbs" at http://theeditorsblog.net/2016/02/21/a-tale-of-adverbs-and-the-comma/

I fixed the "it is".